File: Practica1/Julia/ejercicio2.jl
#= -------------------------------------------------------------------------
# @file ejercicio2.jl
#
# @date 08/03/19 14:50:45
# @author Martin Noblia
# @email mnoblia@disroot.org
#
# @brief
#
# @detail
#
#  Licence:
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or (at
# your option) any later version.
#
# This program is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
---------------------------------------------------------------------------=#
using Images, PyPlot
# Ejercicio 2
# Mostrar el efecto del ruido producido por en una imagen que se toma con bajas
# condiciones de luminosidad

# imagen obtenida con bajo nivel de iluminacion
path3 = "../Imagenes/sci-3.bmp"
img = load(path3)
row_37 = channelview(img[37, :])
row_37_red = row_37[1, :]

# ya que la iluminacion es deficiente vamos a demostrar que el ruido afecta a la medicion del ccd
plot(row_37_red)

File: Practica1/Python/ejercicio2.py
#! /usr/bin/env python

# -*- coding: utf-8 -*-
"""
Created on Tue Jun 25 23:38:05 2013

@author: elsuizo
"""
#Ejercicio 2
#Mostrar el efecto del ruido producido por en una imagen que se toma con bajas
#condiciones de luminosidad

#Imports
#******************************************************************************
import cv2
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.image as mpimg
#******************************************************************************

#imagen sacada con bajo nivel de iluminacion
path3 = '../Imagenes/sci-4.bmp'

#vamos a demostrar como se amplifica el ruido en el sensor ccd

img3_opencv = cv2.imread(path3, 0)
img3_matplot = mpimg.imread(path3)  # leemos con matplotlib(python puro)

#extraigo toda la fila 37
row_37 = img3_opencv[37, :]  # Slicing

#plots
#******************************************************************************
fig, array_ax = plt.subplots(1, 3)
array_ax[0].plot(row_37)
array_ax[1].imshow(img3_opencv)
array_ax[2].imshow(img3_matplot)
plt.show()
